libraries/cms/application/cms.php

```diff
@@ -114,103 +114,15 @@ public function __construct(JInput $input = null, Registry $config = null, JAppl
 		}
 
 		// Enable sessions by default.
-		if (is_null($this->config->get('session')))
+		if (is_null($this->getContainer()->get('config')->get('session')))
```

Comment:
Mostly for service providers that have configuration items (like the ones in [this gist](https://gist.github.com/mbabker/d384671834744d334fd1) for the issue tracker).  What I've been doing as of late is injecting basically our existing `JConfig` object into the application when it's instantiated via the `$config` var so that the application basically becomes the point of truth for the config and the service provider that I use essentially is a wrapper around `Joomla\Registry\Registry::loadFile()` with a little bit of file loading logic.